packages/core/data-transfer/src/strapi/providers/local-destination/strategies/restore/links.ts

```diff
@@ -16,14 +17,23 @@ export const createLinksWriteStream = (
         const { left, right } = link;
         const query = createLinkQuery(strapi, trx);
 
+        const originalLeftRef = left.ref;
+        const originalRightRef = right.ref;
+
         // Map IDs if needed
-        left.ref = mapID(left.type, left.ref) ?? left.ref;
-        right.ref = mapID(right.type, right.ref) ?? right.ref;
+        left.ref = mapID(left.type, originalLeftRef) ?? originalLeftRef;
+        right.ref = mapID(right.type, originalRightRef) ?? originalRightRef;
 
         try {
           await query().insert(link);
         } catch (e) {
           if (e instanceof Error) {
+            if (e.message.toLowerCase().includes('foreign key constraint')) {
```

Comment:
I know it's better to check error codes but I've seen they were particular to each DB type and even tied to the technology behind the DB type, so the one thing they all had in common was that they all included this.

Reply:
Some databases allow localized error messages, so this might be a good fallback but we should also try to match all the known error codes as well, and then add a comment about why it's done this way.